### PR TITLE
Run test runner tests on CI + fix lint

### LINKED ISSUE #2019: CI checks for rust things
Add CI checks for rust things in protostar repo

### PR DESCRIPTION
**Pull request by cptartur**
Closes #2019 

Related cairo1 PR https://github.com/software-mansion-labs/cairo/pull/85

### Checklist
- [ ] Relevant issue is linked
- [ ] Docs updated/issue for docs created
- [ ] Added relevant tests

### CODE REVIEW

**Review by THenry14 — APPROVED** [review]
LGTM :+1:

File: .circleci/config.yml [hunk 1]
```diff
@@ -224,6 +224,20 @@ commands:
...
+      - run:
+          name: Run format check
+          command: cargo fmt --check
+          working_directory: ./protostar-rust
+      - run:
+          name: Run lint check
+          command: cargo clippy --all-targets --all-features -- --no-deps -W clippy::pedantic -A clippy::missing_errors_doc -A clippy::missing_panics_doc -A clippy::default_trait_access
+          working_directory: ./protostar-rust
+          environment:
+            RUSTFLAGS: -Dwarnings
```
Comment (piotmag769) [inline .circleci/config.yml:239]: What does passing it to the compiler do in this case?

**Review by piotmag769 — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (cptartur) [reply .circleci/config.yml:239]: It makes the check fail even on clippy warnings.

**Comment by piotmag769** [comment]
Actually I realised that we linked `rust_runner_tests-linux` to `verify_valid_version_branch` in CI, but not to `patch_test_release_approval_status` and `send_test_release_approval`. Is that intended? I think we should link it to all of the mentioned jobs

**Pull request merged**